plugin.audio.cbcradio/addon.py

```diff
@@ -0,0 +1,121 @@
+import sys
+import urllib
+import urllib2
+import urlparse
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import re
+
+from resources.lib import CBCJsonParser
+
+# TODO Add logging info
+# getting addon object
+my_addon = xbmcaddon.Addon('plugin.audio.cbcradio')
+# Localization info
+language = my_addon.getLocalizedString
+
+# base url and handle of the addon
+base_url = sys.argv[0]
+addon_handle = int(sys.argv[1])
+args = urlparse.parse_qs(sys.argv[2][1:])
+
+# On first time launch, users can select their region, we use a String comparison (not boolean) since that is what getSetting returns
+if my_addon.getSetting('3') == 'true':
+    my_addon.openSettings()
+    my_addon.setSetting('3', 'false')
+
+# Get region setting for Radio 1
+my_r1_region = my_addon.getSetting('0')
+
+# Get region setting for Radio 2
+my_r2_region = my_addon.getSetting('1')
+
+# Get Quality setting
+my_quality = my_addon.getSetting('2').decode('utf-8')
+
+# 30004 is high quality
+if my_quality == language(30004):
+    qual = 0
+else:
+    qual = 1
+
+xbmcplugin.setContent(addon_handle, 'songs')
+
+def build_url(query):
+    return base_url + '?' + urllib.urlencode(query)
+
+mode = args.get('mode',None)
+
+# Top-level menu
+# FIXME, prettify this ugly copy/paste section
+if mode is None:
+
+    url = CBCJsonParser.parse_pls(CBCJsonParser.get_R1_streams(my_r1_region)[qual])
+    li = xbmcgui.ListItem('Radio 1 (' + my_r1_region + ')', iconImage='DefaultAudio.png')
+    li.setProperty('fanart_image',my_addon.getAddonInfo('fanart'))
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li)
+
+    url = CBCJsonParser.parse_pls(CBCJsonParser.get_R2_streams(my_r2_region))
+    li = xbmcgui.ListItem('Radio 2 (' + my_r2_region + ')', iconImage='DefaultAudio.png')
+    li.setProperty('fanart_image',my_addon.getAddonInfo('fanart'))
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li)
+
+    url = CBCJsonParser.parse_pls('http://playerservices.streamtheworld.com/pls/CBC_R3_WEB.pls')
```

Comment:
They changed the website recently, so I don't know where they are now but those were the original links.
When I last tested them yesterday, they seemed to be working.
From [archive.org](https://web.archive.org/web/20140418014428/http://music.cbc.ca/#/streams.aspx)